server/vulnerabilities/macoffice/analyzer.go

```diff
@@ -178,12 +180,12 @@ func Analyze(
 
 		detected := collectVulnerabilities(software, product, relNotes)
 		// The 'software' instance we get back from the iterator does not include vulnerabilities...
-		existing, err := getStoredVulnerabilities(ctx, ds, software.ID)
+		existing, err := getStoredVulnerabilities(ctx, ds.SoftwareByID, software.ID)
```

Comment:
I don't like the idea of passing function pointers of an object.
I understand you want to mock the function easily in the test, but maybe create a dummy object with those mock functions instead?

What concerns me is that it's kind of breaking the object oriented concept here. Instead of passing the object, you make the caller more "knowledgeable of the objects' functions.

Tell me what you think.

Reply:
Thanks, I agree.

Addressed in https://github.com/fleetdm/fleet/pull/10301/commits/1d763092409617af5e52bccf284c46765fe3ff14